flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/tier/remote/RemoteStorageScannerAvailabilityAndPriorityHelper.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid.tiered.tier.remote;
+
+/**
+ * {@link RemoteStorageScannerAvailabilityAndPriorityHelper} is used to help the {@link
+ * RemoteStorageScanner} to notify the availability and priority of the reading process for the
+ * specific partition and subpartition on remote storage.
+ */
+public interface RemoteStorageScannerAvailabilityAndPriorityHelper {
```

Comment:
Looks like this is identical to `NettyConnectionReaderAvailabilityAndPriorityHelper`, and even their implementations `SingleInputGate` are identical. They should be deduplicated